### ISSUE TITLE
Featurerequest: Backgroundlighting

### ISSUE THREAD

**Issue by DVEntertainment** [Winston-Lu/ESP8266-LED-Shelf#9]
Hi! 
I'm planning on haning my clock an inch from the wall, and hanging a ledstrip behind it. That way it will fadely light up the wall behind it in the dark. Could this be controlled with the same webapp? Just another color/brightness selector, and possebly even the effects?

Thanx!
I said it before, but i really like your project, its very stable on my clock!

**Comment by Winston-Lu** [comment]
Hm, it would be possible but I would need to add quite a bit of code for this to work. I can make it work on another pin on the esp8266, but I won't have time to work on it until near the end of this month after exams. I won't have a setup that supports this, so it'll probably be buggy during development. Porting over the effects shouldn't be too difficult or pose much of an issue though. Not going to make any promises on deadlines, but I'll update this thread when I get to it

**Comment by DVEntertainment** [comment]
Thanks! 
No worries, i have it connected to a separate Arduino now. Take your time!

I have another request. Is there a possibility that you could build home assistant integration? Maybe just the powerbutton, but that way I can start the clock just before my alarm goes off etc.

![IMG_20220418_234015](https://user-images.githubusercontent.com/65161300/163949092-a4e8d639-aae6-462f-b4b1-a31dd3eecce9.jpg)

**Comment by Winston-Lu** [comment]
Looks great! I think in the meantime, you could use the typical HTTP request like say make your home assistant visit `http://IP_ADDRESS/?power=1` to turn on, `power=0` for off. I believe someone else is using it like that for smart home integration. I haven't documented what all HTTP commands you can set, but I believe you can take a look in `webserver.cpp` to see what you could interface with

**Comment by Winston-Lu** [comment]
I pushed a new commit with the new code. You'll need to re-upload the web server with LittleFS and change around a few new things in `Config.h` to get it setup with your wiring. Documentation on the home AI HTTP requests has also been put in the README.md file, alongside the changes you would need to make. Rain and Fire aren't supported yet, just wanted to see how well the first push would work first.

**Comment by OniLink57** [comment]
Hello,

For Home Assistant, I added the whole clock to it to replace the webapp and have everything in HA, except the commands. It looks like this : https://ibb.co/dsTqM7K 
For the clock, every number represent a digit from left to right and the brightness can be controlled + the colors. I can also change the effects.
Same for every spot and background.

It took me a WHILE to integrate everything and to understand how to do it but now it fully works.
I used it to automate the clock brightness (brighter for the day and darker for night). The clock also shuts down when I'm in bed or away from home.
